pkg/platform/api/buildplanner/model/buildexpression.go

```diff
@@ -24,12 +28,22 @@ const (
 	OperationRemoved
 	OperationUpdated
 
-	SolveFuncName           = "solve"
-	SolveLegacyFuncName     = "solve_legacy"
-	RequirementsKey         = "requirements"
-	AtTimeKey               = "at_time"
-	RequirementNameKey      = "name"
-	RequirementNamespaceKey = "namespace"
+	SolveFuncName                     = "solve"
+	SolveLegacyFuncName               = "solve_legacy"
+	RequirementsKey                   = "requirements"
+	AtTimeKey                         = "at_time"
+	RequirementNameKey                = "name"
+	RequirementNamespaceKey           = "namespace"
+	RequirementVersionRequirementsKey = "version_requirements"
+	RequirementVersionKey             = "version"
+	RequirementComparatorKey          = "comparator"
+
+	ctxLet         = "let"
+	ctxIn          = "in"
+	ctxAp          = "ap"
+	ctxValue       = "value"
+	ctxAssignments = "assignments"
+	ctxIsAp        = "isAp"
```

Comment:
I feel like the logic for walking the AST should not be part of the model package. The model package should be about marshalling/unmarshalling the data at a high level.

Compare it to for example the json or yaml packages. Both these packages are inevitably used from inside a model package, but they aren't themselves models.

So please keep the model/buildexpression package limited to high level boilerplate for packing and unpacking the data retrieved from API's, but find a more appropriate place for the new AST walking logic.